src/server/req_message.c

```diff
@@ -287,51 +258,21 @@ req_relnodesjob(struct batch_request *preq)
 		return;
 
 	/*
-	 ** Returns job pointer for singleton job or "parent" of
-	 ** an array job.
+	 ** Returns job pointer for singleton job and "parent" of
+	 ** an array sub job.
 	 */
-	pjob = chk_job_request(jid, preq, &jt);
-	if (pjob == NULL) {
+	if (FALSE == chk_job_request(jid, preq, &jt, &js, &pjob, NULL, NULL)) {
 		return;
 	}
 
-	if (jt == IS_ARRAY_NO) {		/* a regular job is okay */
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {		/* a regular job or a subjob is okay */
 		/* the job must be running */
 		if ((pjob->ji_qs.ji_state != JOB_STATE_RUNNING) ||
 			(pjob->ji_qs.ji_substate !=
 			JOB_SUBSTATE_RUNNING)) {
 			req_reject(PBSE_BADSTATE, 0, preq);
 			return;
```

Comment:
I think PBSE_UNKJOBID also need to be handled here.

Reply:
It is handled inside chk_job_request()